PalasoUIWindowsForms/ImageToolbox/ArtOfReadingChooser.cs

```diff
@@ -77,33 +86,32 @@ private void _searchButton_Click(object sender, EventArgs e)
 			try
 			{
 				_thumbnailViewer.Clear();
-				if (!string.IsNullOrEmpty(_searchTermsBox.Text))
+				if (!string.IsNullOrWhiteSpace(_searchTermsBox.Text))
 				{
 					bool foundExactMatches;
-					IEnumerable<object> results = _imageCollection.GetMatchingPictures(_searchTermsBox.Text, out foundExactMatches);
-					if (results.Count() == 0)
-					{
-						_messageLabel.Visible = true;
-						_searchResultStats.Text = "Found no matching images".Localize("ImageToolbox.NoMatchingImages");
-					}
-					else
+					// (avoid enumerating the returned IEnumerable<object> more than once by copying to a List.)
+					var results = new List<object>(_imageCollection.GetMatchingPictures(_searchTermsBox.Text, out foundExactMatches));
+					if (results.Any())
 					{
 						_messageLabel.Visible = false;
+						betterLinkLabel1.Visible = false;
 						_thumbnailViewer.LoadItems(_imageCollection.GetPathsFromResults(results, true));
-						_searchResultStats.Text = string.Format("Found {0} images", results.Count());
+						var fmt = "Found {0} images".Localize("ImageToolbox.MatchingImages", "The {0} will be replaced by the number of matching images");
 						if (!foundExactMatches)
-							_searchResultStats.Text += string.Format(" with names close to {0}.", _searchTermsBox.Text);
+							fmt = "Found {0} images with names close to {1}".Localize("ImageToolbox.AlmostMatchingImages", "The {0} will be replaced by the number of images found.  The {1} will be replaced with the search string.");
+						_searchResultStats.Text = string.Format(fmt, results.Count, _searchTermsBox.Text);
+					}
+					else
+					{
+						_messageLabel.Visible = true;
+						if (PlatformUtilities.Platform.IsWindows && !_searchLanguageMenu.Visible)
```

Comment:
Without knowing what betterLinkLabel1 contains, this is hard to evaluate, but it seems odd that it gets hidden on all platforms but only shown on Windows. Is it an optional component that doesn't WORK on Linux? A comment might be appropriate.